src/LibraryManager/Manifest.cs

```diff
@@ -216,44 +230,58 @@ public Manifest Clone()
                 DestinationPath = destination
             };
 
-            UpdateLibraryProviderAndDestination(desiredState, DefaultProvider, DefaultDestination);
-            if (!desiredState.IsValid(out IEnumerable<IError> errors))
-            {
-                return new List<ILibraryOperationResult> { new LibraryOperationResult(desiredState, errors.ToArray()) };
-            }
+            return await _librariesValidator.ValidateLibraryAsync(desiredState, cancellationToken).ConfigureAwait(false);   
+        }
 
-            IProvider provider = _dependencies.GetProvider(desiredState.ProviderId);
-            if (provider == null)
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="libraryId"></param>
+        /// <param name="providerId"></param>
+        /// <param name="files"></param>
+        /// <param name="destination"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        public async Task<IEnumerable<ILibraryOperationResult>> InstallLibraryAsync(string libraryId, string providerId, IReadOnlyList<string> files, string destination, CancellationToken cancellationToken)
```

Comment:
>IEnumerable<ILibraryOperationResult> [](start = 26, length = 36)

Why does installing a single library now return multiple results?

Reply:
It has validation details for each individual library in the manifest. We can just return one result if any fails. We can either combine all libraries errors collections in one or just return a simple error message. Prafull/Bhavya, what do you need. UI and CLI are the only ones calling it. 

---
In reply to: [195533328](https://github.com/aspnet/LibraryManager/pull/164#discussion_r195533328) [](ancestors = 195533328)